domain/application/state/charm_test.go

```diff
@@ -312,11 +380,20 @@ func (s *charmStateSuite) TestReserveCharmRevision(c *gc.C) {
 	id := charmtesting.GenCharmID(c)
 
 	err := s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
-		_, err := tx.ExecContext(ctx, `INSERT INTO charm (uuid, name, run_as_id) VALUES (?, 'ubuntu', 0)`, id.String())
-		c.Assert(err, jc.ErrorIsNil)
+		_, err := tx.ExecContext(ctx, `INSERT INTO charm (uuid, available) VALUES (?, false)`, id.String())
+		if err != nil {
+			return errors.Trace(err)
+		}
 
-		_, err = tx.ExecContext(ctx, `INSERT INTO charm_state (charm_uuid, available) VALUES (?, false)`, id.String())
-		c.Assert(err, jc.ErrorIsNil)
+		_, err = tx.ExecContext(ctx, `INSERT INTO charm_metadata (charm_uuid, name, run_as_id) VALUES (?, 'ubuntu', 0)`, id.String())
+		if err != nil {
+			return errors.Trace(err)
+		}
+
+		_, err = tx.ExecContext(ctx, `INSERT INTO charm_origin (charm_uuid, reference_name) VALUES (?, 'ubuntu')`, id.String())
+		if err != nil {
+			return errors.Trace(err)
+		}
```

Comment:
This changes a lot because we're not "actively" working on the application. I think you're right, I just want the application service/state to settle down more.